src/PowerShellEditorServices.Hosting/EditorServicesConfig.cs

```diff
@@ -0,0 +1,140 @@
+using System.Management.Automation.Host;
+
+namespace Microsoft.PowerShell.EditorServices.Hosting
+{
+    /// <summary>
+    /// Describes the desired console REPL for the integrated console.
+    /// </summary>
+    public enum ConsoleReplKind
+    {
+        /// <summary>No console REPL - there will be no interactive console available.</summary>
+        None = 0,
+        /// <summary>Use a REPL with the legacy readline implementation. This is generally used when PSReadLine is unavailable.</summary>
+        LegacyReadLine = 1,
+        /// <summary>Use a REPL with the PSReadLine module for console interaction.</summary>
+        PSReadLine = 2,
+    }
+
+    /// <summary>
+    /// Configuration for editor services startup.
+    /// </summary>
+    public sealed class EditorServicesConfig
+    {
+        /// <summary>
+        /// Create a new editor services config object,
+        /// with all required fields.
+        /// </summary>
+        /// <param name="hostInfo">The host description object.</param>
+        /// <param name="psHost">The PowerShell host to use in Editor Services.</param>
+        /// <param name="sessionDetailsPath">The path to use for the session details file.</param>
+        /// <param name="bundledModulePath">The path to the modules bundled with Editor Services.</param>
+        /// <param name="logPath">The path to be used for Editor Services' logging.</param>
+        public EditorServicesConfig(
+            HostInfo hostInfo,
+            PSHost psHost,
+            string sessionDetailsPath,
+            string bundledModulePath,
+            string logPath)
+        {
+            HostInfo = hostInfo;
+            PSHost = psHost;
+            SessionDetailsPath = sessionDetailsPath;
+            BundledModulePath = bundledModulePath;
+            LogPath = logPath;
+        }
+
+        /// <summary>
+        /// The host description object.
+        /// </summary>
+        public HostInfo HostInfo { get; }
+
+        /// <summary>
+        /// The PowerShell host used by Editor Services.
+        /// </summary>
+        public PSHost PSHost { get; }
+
+        /// <summary>
+        /// The path to use for the session details file.
+        /// </summary>
+        public string SessionDetailsPath { get; }
+
+        /// <summary>
+        /// The path to the modules bundled with EditorServices.
+        /// </summary>
+        public string BundledModulePath { get; }
+
+        /// <summary>
+        /// The path to use for logging for Editor Services.
+        /// </summary>
+        public string LogPath { get; }
+
+        /// <summary>
+        /// Names of or paths to any additional modules to load on startup.
+        /// </summary>
+        public IReadOnlyList<string> AdditionalModules { get; set; } = null;
+
+        /// <summary>
+        /// Flags of features to enable on startup.
+        /// </summary>
+        public IReadOnlyList<string> FeatureFlags { get; set; } = null;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this initialization to 'FeatureFlags', the compiler will do that for you.](https://app.codacy.com/app/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4579423)